picfixPy/sharpen.py

```diff
@@ -1,20 +1,112 @@
-def sharpen(input_img, intensity=5, output_img=''):
+# dependencies
+import numpy as np
+import skimage.io
+
+
+def sharpen(input_img, intensity=5, display=False, output_img=''):
     
     '''
-    Sharpens an image, given the intensity specificication
+    Sharpens an image with given intensity specification
+    
     
     Input
     ============
-    input_img -- string of path for an image file in .png format
+    input_img -- string of path for an image file in .jpg, .jpeg, .png format
     intensity -- integer in range [0,10] to specificy intensity of sharpening applied to the image,
                  defaults to 5.
-    output_img -- string path for sharpened .png image
+    display -- bool. If True, output will be displayed. Defaults to False.
+    output_img -- string path for output of corresponding .jpg, .jpeg, .png image with adjusted sharpening 
     
     Returns
     ============
-    an sharpened image file in .png format in specified output path
+    an image file in .jpg, .jpeg, .png format with adjusted sharpening intensity in specified output path
     
     Example
     ============
-    sharpen("../img/image.png", 2, "../img/sharpened_image.png")
-    '''
\ No newline at end of file
+    sharpen("../img/image.jpg", 5, display=False, "../img/sharpened_image.jpg")
+    '''
+
+    # load input image
+    # lift pixel value restrictions between 0 and 255 temporarily
+    try: 
+        input_img = skimage.io.imread(input_img).astype(int)
+    
+    except AttributeError:
+        print("Please use a string to specify the file path for an input image.")
+        raise
+    
+    except OSError:
+        print("Please provide an image file.")
+        raise
+    
+    except FileNotFoundError:
+        print("Cannot find image file.")
+            
+    except Exception as error:
+        print("Cannot load image, something went wrong :(")
+        print(error)
+        raise
+        
+    # validate intensity level
+    if (intensity < 0 or intensity > 10):
+        raise ValueError("Intensity level must be between 0 and 10.")
+        
+        
+    # Guassian blurring of image
+    
+    # approximation of Gaussian blur 3x3 kernel
+    # kernel attribution: https://en.wikipedia.org/wiki/Kernel_(image_processing)
+    kernel = np.array([[1.0,2.0,1.0], [2.0,4.0,2.0], [1.0,2.0,1.0]])
+    kernel = kernel / np.sum(kernel)
+    
+    # attribution: https://bit.ly/2EbS4XD
```

Comment:
It's important that we give credits where credits are due, good job attributing the author of this blurring approach!

Reply:
Thanks